From cb4ed12ffc332d1f72d054ed92655b5f1c38f621 Mon Sep 17 00:00:00 2001
From: Darren Tucker <dtucker@dtucker.net>
Date: Sat, 19 Aug 2023 07:39:08 +1000
Subject: [PATCH] Fix zlib version check for 1.3 and future version.

bz#3604.
---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Origin: upstream, https://github.com/openssh/openssh-portable/commit/cb4ed12ffc332d1f72d054ed92655b5f1c38f621
Bug: https://bugzilla.mindrot.org/show_bug.cgi?id=3604
Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/openssh/+bug/2049552
Last-Update: 2024-01-17
--- a/configure.ac
+++ b/configure.ac
@@ -1488,7 +1488,7 @@
 	[[
 	int a=0, b=0, c=0, d=0, n, v;
 	n = sscanf(ZLIB_VERSION, "%d.%d.%d.%d", &a, &b, &c, &d);
-	if (n != 3 && n != 4)
+	if (n < 1)
 		exit(1);
 	v = a*1000000 + b*10000 + c*100 + d;
 	fprintf(stderr, "found zlib version %s (%d)\n", ZLIB_VERSION, v);
