From fcd78e31cdd45a7e69ccfe6d8a3b1037dc1de290 Mon Sep 17 00:00:00 2001
From: "djm@openbsd.org" <djm@openbsd.org>
Date: Wed, 24 May 2023 23:01:06 +0000
Subject: [PATCH] upstream: fix AuthorizedPrincipalsCommand when
 AuthorizedKeysCommand

Origin: upstream, https://github.com/openssh/openssh-portable/commit/fcd78e31cdd45a7e69ccfe6d8a3b1037dc1de290
Bug: https://bugzilla.mindrot.org/show_bug.cgi?id=3574
Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/openssh/+bug/2031942
Last-Update: 2023-08-24
---
 servconf.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/servconf.c b/servconf.c
index 9d5b73ecfc0..0f4fd2f17e3 100644
--- a/servconf.c
+++ b/servconf.c
@@ -1,5 +1,5 @@
 
-/* $OpenBSD: servconf.c,v 1.392 2023/03/05 05:34:09 dtucker Exp $ */
+/* $OpenBSD: servconf.c,v 1.393 2023/05/24 23:01:06 djm Exp $ */
 /*
  * Copyright (c) 1995 Tatu Ylonen <ylo@cs.hut.fi>, Espoo, Finland
  *                    All rights reserved
@@ -2403,7 +2403,7 @@ process_server_config_line_depth(ServerOptions *options, char *line,
 			fatal("%.200s line %d: %s must be an absolute path",
 			    filename, linenum, keyword);
 		}
-		if (*activep && options->authorized_keys_command == NULL)
+		if (*activep && *charptr == NULL)
 			*charptr = xstrdup(str + len);
 		argv_consume(&ac);
 		break;
