Description: Mention ssh-keygen in ssh fingerprint changed warning
Author: Scott Moser <smoser@ubuntu.com>
Bug: https://bugzilla.mindrot.org/show_bug.cgi?id=1843
Bug-Ubuntu: https://bugs.launchpad.net/bugs/686607
Last-Update: 2010-12-14

--- a/sshconnect.c
+++ b/sshconnect.c
@@ -908,14 +908,17 @@
 			error("%s. This could either mean that", key_msg);
 			error("DNS SPOOFING is happening or the IP address for the host");
 			error("and its host key have changed at the same time.");
-			if (ip_status != HOST_NEW)
+			if (ip_status != HOST_NEW) {
 				error("Offending key for IP in %s:%d", ip_file, ip_line);
+				error("  remove with: ssh-keygen -f \"%s\" -R %s", ip_file, ip);
+			}
 		}
 		/* The host key has changed. */
 		warn_changed_key(host_key);
 		error("Add correct host key in %.100s to get rid of this message.",
 		    user_hostfile);
 		error("Offending key in %s:%d", host_file, host_line);
+		error("  remove with: ssh-keygen -f \"%s\" -R %s", host_file, host);
 
 		/*
 		 * If strict host key checking is in use, the user will have
